src/compiler/parser.ts

```diff
@@ -4049,6 +4049,15 @@ namespace ts {
             node.expression = allowInAnd(parseExpression);
             parseExpected(SyntaxKind.CloseParenToken);
             node.thenStatement = parseStatement();
+            
+            if (node.thenStatement.kind === SyntaxKind.EmptyStatement) {

```

Comment:
do not do this in the parser. this should be done in the checker.ts, possibly in a new function called checkgrammarIfStatement, see other checkGrammar* functions.

The reason doing it in the parser is not correct, is it will break assumptions about node reuse, which the parser does when it is incrementally parsing. you want to make sure that the error is always reported correctly, in this case it will not be if you add a semicolon then remove it.